tests/api2/test_openssl.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from middlewared.test.integration.utils import call, ssh
+from auto_config import ha
+
+retry = 5
+fips_version = "3.0.9"
+
+
+# Sometimes this test fails because the testing environment has broken failover (randomly. Fun transient error. Reports a failed heartbeat).
+@pytest.mark.flaky(reruns=retry, reruns_delay=5)
+@pytest.mark.skipif(not ha, reason='Test only valid for HA')
+def test_fips_version():
+    # The reason we have a set of commands in a payload is because of some annoying FIPS technicalities.
+    # Basically, when FIPS is enabled, we can't use SSH because the SSH key used by root isn't using a FIPS provided algorithm. (this might need to be investigated further)
```

Comment:
Btw you should be able to pull that off with `-c aes128-ctr` iirc flag (not saying it's necessary but just sharing here)